### PR TITLE
Add OAuth Client name to the login title

### PR DESCRIPTION
**Pull request by scruffian**
This pull request updates the header text on the login page when a client id is set, to display the name of the OAuth2 client application.
  
#### Testing instructions
  
1. Run `git checkout add/oauth-title` and start your server
2. Open the `Login` page with a client id - http://calypso.localhost:3000/log-in?clientId=1
3. Check that you see the name of the client application in the header.
4. Try a different client id and check that you still see the same name 

#### Reviews
  
- [x] Code
- [ ] Product

### CODE REVIEW

File: client/login/controller.js [hunk 1]
```diff
@@ -12,15 +12,16 @@ import HandleEmailedLinkForm from './magic-login/handle-emailed-link-form';
...
 
 		context.primary = (
 			<WPLogin
 				locale={ lang }
 				path={ path }
 				twoFactorAuthType={ twoFactorAuthType }
 				socialConnect={ flow === 'social-connect' }
-				privateSite={ flow === 'private-site' } />
+				privateSite={ flow === 'private-site' }
+				clientId={ query.clientId } />
```
Comment (scruffian) [inline client/login/controller.js]: I wonder if the query string should be `oauth2ClientId` or `oauth2-client-id`?

Comment (jsnmoon) [inline client/login/controller.js]: Hmm... Some programming languages force case insensitivity on query strings, but we don't have a standard about this as far as I can tell.

(FWIW, RFC 3986 says URIs are should be case sensitive except for the scheme and the host.)

Comment (pento) [inline client/login/controller.js]: Historically, WP.com has used lowercase, and underscores to separate words. Ie, `client_id`.

Comment (pento) [inline client/login/controller.js]: For a super lazy reference:

```
$ find . -name 'controller.js' | xargs ack 'query\.[a-z]+\b' | wc -l
      31
$ find . -name 'controller.js' | xargs ack 'query\.[a-z]+_[a-z_]*\b' | wc -l
       3
$ find . -name 'controller.js' | xargs ack 'query\.[a-z]+-[a-z-]*\b' | wc -l
       0
$ find . -name 'controller.js' | xargs ack 'query\.[a-z]+[A-Z][a-zA-Z]*\b' | wc -l
       2
```

So, underscore or camelCase seem to be the options, hyphen is never used.

**Comment by pento** [comment]
Oh, I just remembered that we use `client_id` in all our public [docs](https://developer.wordpress.com/docs/oauth2/). Let's not change that.

**Pull request merged**